cipher_suite.go

```diff
@@ -69,6 +71,9 @@ type CipherSuite interface {
 	// AuthenticationType controls what authentication method is using during the handshake
 	AuthenticationType() CipherSuiteAuthenticationType
 
+	// UsesEllipticCurves determines whether elliptic curves will be used in the client hello message.
+	UsesEllipticCurves() bool
```

Comment:
Should this be `SendEllipticCurvesInClientHello`? Or is this really communicating `Use EcdhePSKPreMasterSecret when generating preMasterSecret`

Would it make sense to add `GeneratePreMasterSecret` to the CipherSuite interface?

Reply:
Refactored, introduced `KeyExchangeAlgorithm()` which is more complaint with [RFC4279](https://datatracker.ietf.org/doc/html/rfc4279)